app/models/vote_account.rb

```diff
@@ -32,6 +32,10 @@ def vote_account_history_last
     vote_account_histories.last
   end
 
+  def vote_account_history_last_in_batch(batch_uuid)
+    vote_account_histories.where(batch_uuid: batch_uuid).last
```

Comment:
There should be only one vote_account_history for the given batch. I think you can remove `.last` since that will only slow down the results.

Reply:
Rename to `vote_account_history_for(batch_uuid)`